src/explore-education-statistics-frontend/src/services/tableHeadersService.ts

```diff
@@ -0,0 +1,39 @@
+import sortBy from 'lodash/sortBy';
+import { Dictionary } from '@common/types/util';
+import {
+  CategoryFilter,
+  Indicator,
+  LocationFilter,
+} from '@frontend/modules/table-tool/components/types/filters';
+import TimePeriod from '@frontend/modules/table-tool/components/types/TimePeriod';
+
+const removeSiblinglessTotalRows = (
+  categoryFilters: Dictionary<CategoryFilter[]>,
+): CategoryFilter[][] => {
+  return Object.values(categoryFilters).filter(filter => {
+    return filter.length > 1 || !filter[0].isTotal;
+  });
+};
+
+const getDefaultTableHeaderConfig = (
+  indicators: Indicator[],
+  filters: Dictionary<CategoryFilter[]>,
+  timePeriods: TimePeriod[],
+  locations: LocationFilter[],
+) => {
+  const sortedFilters = sortBy(
+    [...removeSiblinglessTotalRows(filters), locations],
+    [options => options.length],
+  );
+
+  const halfwayIndex = Math.floor(sortedFilters.length / 2);
+
+  return {
+    columnGroups: sortedFilters.slice(0, halfwayIndex),
+    rowGroups: sortedFilters.slice(halfwayIndex),
+    columns: timePeriods,
+    rows: indicators,
+  };
+};
+
+export default getDefaultTableHeaderConfig;
```

Comment:
I wouldn't call this a 'service' as 'service' is the convention used for code that actually accesses external services e.g. web APIs, databases, etc.

In this case, we're actually just talking about a utility function that is highly specialised to table tool. Remember I've said that common (or more generalised) modules shouldn't really be importing modules from more nested parts of the project  :wink:

Consequently, in this instance, the place I would be least surprised to find this would be in the `table-tool` module under `utils`.

Reply:
Makes perfect sense, thanks!